com.hazelcast.internal.cluster.impl.BasicClusterStateTest.pendingInvocations_shouldBeNotified_whenMemberLeft_whenClusterState_PASSIVE [HZ-1708]

_5.0.z_ (commit 3794b36b4d7e080517d41adf41a0c91abaa22d2d)

Failed on Openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-Openj9-11/15/testReport/com.hazelcast.internal.cluster.impl/BasicClusterStateTest/pendingInvocations_shouldBeNotified_whenMemberLeft_whenClusterState_PASSIVE/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expected: an instance of com.hazelcast.core.MemberLeftException
     but: <java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Cluster is in PASSIVE state! Operation: com.hazelcast.partition.IndeterminateOperationStateExceptionTest$SilentOperation{serviceName='null', identityHash=-476619532, partitionId=-1, replicaIndex=0, callId=3, invocationTime=1641177075445 (2022-01-03 02:31:15.445), waitTimeout=-1, callTimeout=60000, tenantControl=com.hazelcast.spi.impl.tenantcontrol.NoopTenantControl@0}> is a java.util.concurrent.ExecutionException
Stacktrace was: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Cluster is in PASSIVE state! Operation: com.hazelcast.partition.IndeterminateOperationStateExceptionTest$SilentOperation{serviceName='null', identityHash=-476619532, partitionId=-1, replicaIndex=0, callId=3, invocationTime=1641177075445 (2022-01-03 02:31:15.445), waitTimeout=-1, callTimeout=60000, tenantControl=com.hazelcast.spi.impl.tenantcontrol.NoopTenantControl@0}
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.returnOrThrowWithGetConventions(InvocationFuture.java:121)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:100)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:609)
	at com.hazelcast.internal.cluster.impl.BasicClusterStateTest.pendingInvocations_shouldBeNotified_whenMemberLeft_whenClusterState_doesNotAllowJoin(BasicClusterStateTest.java:428)
	at com.hazelcast.internal.cluster.impl.BasicClusterStateTest.pendingInvocations_shouldBeNotified_whenMemberLeft_whenClusterState_PASSIVE(BasicClusterStateTest.java:406)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:866)
Caused by: java.lang.IllegalStateException: Cluster is in PASSIVE state! Operation: com.hazelcast.partition.IndeterminateOperationStateExceptionTest$SilentOperation{serviceName='null', identityHash=-476619532, partitionId=-1, replicaIndex=0, callId=3, invocationTime=1641177075445 (2022-01-03 02:31:15.445), waitTimeout=-1, callTimeout=60000, tenantControl=com.hazelcast.spi.impl.tenantcontrol.NoopTenantControl@0}
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.checkNodeState(OperationRunnerImpl.java:323)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:235)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:469)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
	at ------ submitted from ------.()
	at com.hazelcast.internal.util.ExceptionUtil.cloneExceptionWithFixedAsyncStackTrace(ExceptionUtil.java:279)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.returnOrThrowWithGetConventions(InvocationFuture.java:112)
	... 16 more

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:964)
	at org.junit.Assert.assertThat(Assert.java:930)
	at org.junit.rules.ExpectedException.handleException(ExpectedException.java:271)
	at org.junit.rules.ExpectedException.access$000(ExpectedException.java:111)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:260)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:62)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:49)
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:162)
	at java.base/java.lang.Thread.run(Thread.java:866)
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: test_noMigration_whenNodeLeaves_onClusterState_FROZEN in 7.546 seconds.

No metrics recorded during the test
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/BasicClusterStateTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/BasicClusterStateTest.java
@@ -43,6 +43,7 @@ import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.transaction.TransactionOptions;
 import com.hazelcast.transaction.TransactionOptions.TransactionType;
+import org.junit.Assert;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -424,8 +425,7 @@ public class BasicClusterStateTest extends HazelcastTestSupport {
         changeClusterStateEventually(hz2, state);
         hz1.shutdown();
 
-        expectedException.expect(MemberLeftException.class);
-        future.get();
+        assertTrueEventually(() -> Assert.assertThrows(MemberLeftException.class, future::get));
     }
 
     @Test

